src/enzo/Grid_FDMCollapse.C

```diff
@@ -130,9 +136,162 @@ if(QuantumPressure){
+  int CollapseTestParticleCount = 0;
+  int SetupLoopCount, npart = 0;
+  int ParticleCount = 0;
+  int ind, indxp, indxn, indyp, indyn, indzp, indzn;
+  int ip,in,jp,jn,kp,kn;
+  double x,y,z,vx,vy,vz;
+
+  if (UseParticles > 0){
+    if (ProcessorNumber != MyProcessorNumber) {
+      NumberOfParticles = (UseParticles > 0) ? 1 : 0;
+    for (dim = 0; dim < GridRank; dim++)
+      NumberOfParticles *= (GridEndIndex[dim] - GridStartIndex[dim] + 1);
+    return SUCCESS;
+    }
 
 
+      fprintf(stderr, "initialize particles \n" );
+    for (SetupLoopCount = 0; SetupLoopCount < 1+min(UseParticles, 1); SetupLoopCount++) {
+     if (SetupLoopCount > 0) {
+      /* If particles already exist (coarse particles), then delete. */
+        if (NumberOfParticles > 0) this->DeleteParticles();
+      /* Use count from previous loop to set particle number. */
+        NumberOfParticles = npart;
+        npart = 0;
+      /* Allocate space. */
+        this->AllocateNewParticles(NumberOfParticles);
+      /* Particle values will be set below. */
+      }
+	// set some test particles
+	ParticleCount = 1000;
+
+	// set a binary bh
+	/*npart = 1;
+    if (SetupLoopCount > 0) {
+	double vrot = sqrt(6.67e-6*1e6*2e33/3.1e21)/2./(LengthUnits/TimeUnits);
+    ParticleMass[0] = ParticleMeanDensity;
+	ParticleNumber[0] = CollapseTestParticleCount++;
+    ParticleType[0] = PARTICLE_TYPE_DARK_MATTER;
+    ParticlePosition[0][0] = 0.525 ;
+	ParticlePosition[1][0] = 0.5 ;
+	ParticlePosition[2][0] = 0.5 ;
+	ParticleVelocity[0][0] = 0.0;
+    ParticleVelocity[1][0] = 1.0e5 /(LengthUnits/TimeUnits);
+    ParticleVelocity[2][0] = 0.0;
+
+    ParticleMass[1] = ParticleMeanDensity;
+    ParticleNumber[1] = CollapseTestParticleCount++;
+	ParticleType[1] = PARTICLE_TYPE_DARK_MATTER;
+    ParticlePosition[0][1] = 0.6 ;
+	ParticlePosition[1][1] = 0.5 - 3.1e21/LengthUnits;
+	ParticlePosition[2][1] = 0.5 ;
+	ParticleVelocity[0][1] = vrot;
+	ParticleVelocity[1][1] = 2.0e6 /(LengthUnits/TimeUnits);
+    ParticleVelocity[2][1] = 0.0;
+	}*/
+	// set many particles
+	while (ParticleCount > 0) {
+        if (SetupLoopCount > 0) {
+		    ParticleMass[npart] = ParticleMeanDensity;
+	        ParticleNumber[npart] = CollapseTestParticleCount++;
+            ParticleType[npart] = PARTICLE_TYPE_DARK_MATTER;
+         // Set random position within cell.
+		    double theta = 3.1415927/6./1000*npart;
+		    ParticlePosition[0][npart] = 0.5 + 1e-1*(FLOAT(rand())/FLOAT(RAND_MAX) - 0.5);
+		    ParticlePosition[1][npart] = 0.5 + 1e-1*(FLOAT(rand())/FLOAT(RAND_MAX) - 0.5);
+		    ParticlePosition[2][npart] = 0.5 + 1e-1*(FLOAT(rand())/FLOAT(RAND_MAX) - 0.5);
+			ParticleVelocity[0][npart] = 0.0;//-2.5e6/(LengthUnits/TimeUnits)*sin(theta);
+			ParticleVelocity[1][npart] =  hmcoef*8*3.1415927;//2.0e6 /(LengthUnits/TimeUnits);//*cos(theta);
+			ParticleVelocity[2][npart] = 0.0;
+		}
+		ParticleCount--;
+		npart++;
+	}
+
+    // set particles following the FDM density
+    /*
```

Comment:
Thanks Greg, I still would like to keep this part, as it sets the test particles according to some density distribution. It could be useful for future uses.